tests/cases/compiler/throwType_function_parameter.ts

```diff
@@ -0,0 +1,11 @@
+function checkParameterPosition<T extends number>(y: T extends 1234 ? throw 'No zero' : T) {
+    y.toExponential()
+}
+checkParameterPosition(1234)
+checkParameterPosition(12345678)
+
+type MustNumber<T> = T extends number ? T : throw `"found "${typeof T}"`
+type MustNumber2<T> = T extends number ? T : throw `"found "${typeof T}"`
```

Comment:
These still use `typeof T` instead of the new intrinsic type.

Reply:
thanks, test updated